crates/common/src/operation.rs

```diff
@@ -2,66 +2,99 @@ use anyhow::{anyhow, Context, Result};
 use base64::{engine::general_purpose::STANDARD as engine, Engine as _};
 use bincode;
 use celestia_types::Blob;
-use ed25519_dalek::{Signature, Signer, SigningKey, VerifyingKey};
+use ed25519_dalek::{
+    Signature as Ed25519Signature, Signer as Ed25519Signer, SigningKey as Ed25519SigningKey,
+    VerifyingKey as Ed25519VerifyingKey,
+};
+use lazy_static::lazy_static;
 use prism_errors::GeneralError;
+use secp256k1::{
+    ecdsa::Signature as Secp256k1Signature, Message as Secp256k1Message,
+    PublicKey as Secp256k1VerifyingKey, Secp256k1, SecretKey as Secp256k1SigningKey,
+};
 use serde::{Deserialize, Serialize};
 use std::{self, fmt::Display};
 
+use crate::tree::Digest;
+
+lazy_static! {
```

Comment:
Idk if this is actually safe, I'm assuming it uses a random number generator and not sure it should be static like this.

Reply:
we can use the global-context feature from secp256k1